Objects/longobject.c

```diff
@@ -6775,3 +6776,121 @@ int PyLong_AsUInt64(PyObject *obj, uint64_t *value)
 {
     LONG_TO_UINT(obj, value, "C uint64_t");
 }
+
+
+static const PyLongLayout PyLong_LAYOUT = {
+    .bits_per_digit = PyLong_SHIFT,
+    .digits_order = -1,  // least significant first
+    .digit_endianness = PY_LITTLE_ENDIAN ? -1 : 1,
+    .digit_size = sizeof(digit),
+};
+
+
+const PyLongLayout*
+PyLong_GetNativeLayout(void)
+{
+    return &PyLong_LAYOUT;
+}
+
+
+int
+PyLong_Export(PyObject *obj, PyLongExport *export_long)
+{
+    if (!PyLong_Check(obj)) {
```

Comment:
Let's export zero on failure, per discussion in https://github.com/capi-workgroup/api-evolution/issues/32.
```suggestion
    if (!PyLong_Check(obj)) {
        export_long->value = 0;
        export_long->digits = NULL;
```
(and perhaps the other ones too.)